man/fi_endpoint.3.md

```diff
@@ -520,6 +520,19 @@ The following option levels and option names and parameters are defined.
   that applications that want to override the default MIN_MULTI_RECV
   value set this option before enabling the corresponding endpoint.
 
+- *FI_OPT_FI_HMEM_P2P - int*
+: Defines how the provider should handle peer to peer FI_HMEM transfers for
+  this endpoint. The default option is to require peer to peer support. Valid
+  values defined in fi_endpoint.h are:
```

Comment:
The default is not to require p2p, but allow the provider to make the decision based on the type transfer.  I would consider these options:
```
Enabled - p2p is permitted, subject to the provider implementation
Required - p2p is required for all hmem transfers.  Transfers that cannot be performed using p2p will be reported as failing.
Preferred - provider should use p2p transfers when possible
Disabled - p2p transfers should not be used
```

Reply:
Is the 'Preferred' option still needed if we have a 'Required' option? It seems like 'Enabled' and 'Preferred' are the same in your example, but I might be missing something here.

Reply:
The difference is how the provider handles cases where p2p is *available*:

Enabled - provider selects when to use p2p.  For example, use p2p for large transfers, but copies for small transfers.
Preferred - provider uses it when possible.  For example, provider will use p2p for large and small transfers.

Both allow for falling back to host copies if the peer device is not usable for p2p transfers.

Reply:
That makes sense, thank you. Let me know if the update works for you.